src/go/rpk/pkg/cli/cluster/partitions/show.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package partitions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/types"
+)
+
+func newShowPartitionAssignmentsCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	type result struct {
+		Topic     string
+		Partition int
+		Leader    int
+		NodeCore  []string
+	}
+	var partitions []int
+	cmd := &cobra.Command{
+		Use:   "show",
+		Short: "Show partition replica assignments with CPU cores",
+		Long:  helpShowAssignments,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+			out.CheckExitCloudAdmin(p)
+
+			cl, err := adminapi.NewClient(fs, p)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			var ns, t string
+			var reqPartitions []int
+			var results []result
+
+			for _, topic := range topics {
+				if s := strings.Split(topic, "/"); len(s) == 2 {
+					ns = s[0]
+					t = s[1]
+				} else if len(s) == 1 {
+					ns = "kafka"
+					t = s[0]
+				} else {
+					fmt.Printf("invalid format: %s, skipping\n", topic)
+					continue
+				}
+
+				if len(partitions) == 0 {
+					allPart, err := cl.GetTopic(cmd.Context(), ns, t)
+					out.MaybeDie(err, "unable to get topic information: %v\n", err)
+					reqPartitions = createPartitionList(len(allPart))
+				} else {
+					reqPartitions = partitions
+				}
+
+				for _, p := range reqPartitions {
+					pInfo, err := cl.GetPartition(cmd.Context(), ns, t, p)
```

Comment:
This is serial -- can it be concurrent? This can probably use the same waitgroup that will guard the comment I made just above.

Reply:
Fixed in https://github.com/redpanda-data/redpanda/pull/14025/commits/75b27f460ee265187d46ff102c76d196e2665212